trade-finance-manager-ui/server/controllers/utilisation-reports/add-payment/index.ts

```diff
@@ -87,21 +99,27 @@ export const addPayment = async (req: AddPaymentRequest, res: Response) => {
     const { reportId } = req.params;
     const checkedCheckboxIds = getPremiumPaymentsCheckboxIdsFromObjectKeys(req.body);
     const feeRecordIds = checkedCheckboxIds.map((checkboxId) => getFeeRecordIdFromPremiumPaymentsCheckboxId(checkboxId));
-    const { errors, formValues, paymentNumber } = extractAddPaymentFormValuesAndValidateIfPresent(req.body);
+    const feeRecordPaymentCurrency = getFeeRecordPaymentCurrencyFromPremiumPaymentsCheckboxId(checkedCheckboxIds[0]);
+
+    const { isAddingPayment, errors, formValues, paymentNumber } = extractAddPaymentFormValuesAndValidateIfPresent(req.body, feeRecordPaymentCurrency);
+    const canSubmitFormValues = errors.errorSummary.length === 0;
 
-    if ('addPaymentFormSubmission' in req.body && errors.errorSummary.length === 0) {
-      // TODO FN-1739 save data upon valid submission
-      return res.status(501).send('Your request was valid, but we have not finished building the feature');
+    if (isAddingPayment && canSubmitFormValues) {
+      const { addAnotherPayment, ...paymentFormValues } = formValues;
+      await api.addPaymentToFeeRecords(reportId, paymentFormValues, feeRecordIds, user, userToken);
+      if (addAnotherPayment && addAnotherPayment !== 'true') {
```

Comment:
is the radio allowed to be unticked and still submit the form? Assumed you'd have to pick yes or no and not have it as undefined

Reply:
This is more of a typescript thing as the type at this point in the code has been asserted to be `'true' | 'false'`, but it would be hard to make it so that typescript picks that up. I have added a `ValidatedAddPaymentFormValues` type so that I can make this check more clear - let me know what you think